dask/dataframe/csv.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import print_function, division, absolute_import
+
+from io import BytesIO
+from warnings import warn
+
+import numpy as np
+import pandas as pd
+
+from ..delayed import delayed
+from .io import from_delayed
+
+from ..bytes import read_bytes
+from ..bytes.compression import seekable_files, files as cfiles
+from ..utils import ensure_bytes
+
+
+delayed = delayed(pure=True)
+
+
+def bytes_read_csv(b, header, kwargs, dtypes=None, columns=None):
```

Comment:
Do you envisage a dispatch system here too for different filetypes (avro, feather, parquet ...) ?